test/integration/multiple.done.js

```diff
@@ -4,25 +4,95 @@ var args   = [];
 
 describe('multiple calls to done()', function() {
   var res;
-
   this.timeout(1000);
 
-  before(function(done) {
-    run('multiple.done.js', args, function(err, result) {
-      res = result;
-      done(err);
+  describe('from a spec', function() {
+    before(function(done) {
+      run('multiple.done.js', args, function(err, result) {
+        res = result;
+        done(err);
+      });
+    });
+
+    it('results in failures', function() {
+      assert.equal(res.stats.pending, 0);
+      assert.equal(res.stats.passes, 1);
+      assert.equal(res.stats.failures, 1);
+      assert.equal(res.code, 1);
+    });
+
+    it('throws a descriptive error', function() {
+      assert.equal(res.failures[0].err.message,
+        'done() called multiple times');
+    });
+  });
+
+  describe('with multiple specs', function() {
+    before(function(done) {
+      run('multiple.done.specs.js', args, function(err, result) {
+        res = result;
+        done(err);
+      });
+    });
+
+    it('results in a failure', function() {
+      assert.equal(res.stats.pending, 0);
+      assert.equal(res.stats.passes, 2);
+      assert.equal(res.stats.failures, 1);
+      assert.equal(res.code, 1);
+    });
+
+    it('correctly attributes the error', function() {
+      assert.equal(res.failures[0].fullTitle, 'suite test1');
+      assert.equal(res.failures[0].err.message,
+        'done() called multiple times');
     });
   });
 
-  it('results in failures', function() {
-    assert.equal(res.stats.pending, 0);
-    assert.equal(res.stats.passes, 1);
-    assert.equal(res.stats.failures, 1);
-    assert.equal(res.code, 1);
+  describe('from a before hook', function() {
+    before(function(done) {
+      run('multiple.done.before.js', args, function(err, result) {
+        res = result;
+        done(err);
+      });
+    });
+
+    it('results in a failure', function() {
+      assert.equal(res.stats.pending, 0);
+      assert.equal(res.stats.passes, 1);
```

Comment:
I see. Since (at least in this case) we know that the before hook has failed before the test has finished running, couldn't we ignore the outcome of the test?